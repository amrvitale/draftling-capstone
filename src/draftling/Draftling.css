.individualCard {
  text-align: center;
  padding: 15px;
  margin: 1px;
}
a {
  text-decoration: none;
}

.card-title {
  font-size: 19px;
}

.card-summary {
  font-size: 18px;
}

.card-meta {
  font-size: 16px;
}
.card-header {
  height: 30%;
  color: white;
  background: maroon;
  text-align: center;
}