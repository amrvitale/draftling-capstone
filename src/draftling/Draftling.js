import React from 'react'
import ApiContext from '../ApiContext'
import { Link } from 'react-router-dom'
import './Draftling.css'

export default class Draftling extends React.Component {
    
    static contextType = ApiContext;

  
    render() {
        const { title, id, content, wordcount, genre, modified} = this.props

        const date = new Date(modified);

        return (
            <div className="individualCard">
                <Link to={`/draftling/${id}`}>  
                    <div className="card-header">
                        <div className="card-title">
                            <h2>{title}</h2> 
                        </div>
                    </div>
                    <div className="card-summary">
                        {wordcount},
                        <br />
                        {genre}
                        <br />
                    </div>
                    <div className="card-meta">Created: {date.toDateString()}</div>
                </Link>
            </div>
        )
    }
}


