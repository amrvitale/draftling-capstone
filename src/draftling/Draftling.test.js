import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Draftling from './Draftling'

describe (`Draftling component`, () => {
    const props = {
        id: '13',
        title: 'test-draftling',
        wordcount: 'sixWordStory',
        genre: 'Horror',
        modified: new Date(2020, 6, 17),
    }

    it ('renders a .Draftling by default', () => {
        const wrapper = shallow( <Draftling />)
        expect (toJson(wrapper)).toMatchSnapshot()
    })

    it ('renders the Draftling given props', () => {
        const wrapper = shallow(<Draftling {...props} />)
        expect (toJson(wrapper)).toMatchSnapshot()
    })
});