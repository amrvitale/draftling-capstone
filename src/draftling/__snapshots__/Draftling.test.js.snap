// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Draftling component renders a .Draftling by default 1`] = `
<div
  className="individualCard"
>
  <Link
    to="/draftling/undefined"
  >
    <div
      className="card-header"
    >
      <div
        className="card-title"
      >
        <h2 />
      </div>
    </div>
    <div
      className="card-summary"
    >
      ,
      <br />
      <br />
    </div>
    <div
      className="card-meta"
    >
      Created: 
      Invalid Date
    </div>
  </Link>
</div>
`;

exports[`Draftling component renders the Draftling given props 1`] = `
<div
  className="individualCard"
>
  <Link
    to="/draftling/13"
  >
    <div
      className="card-header"
    >
      <div
        className="card-title"
      >
        <h2>
          test-draftling
        </h2>
      </div>
    </div>
    <div
      className="card-summary"
    >
      sixWordStory
      ,
      <br />
      Horror
      <br />
    </div>
    <div
      className="card-meta"
    >
      Created: 
      Fri Jul 17 2020
    </div>
  </Link>
</div>
`;
